src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -318,7 +335,11 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                 }
             }
 
-            return $reflectionClass->newInstanceArgs($params);
+            if ($constructor->isConstructor()) {
+                return $reflectionClass->newInstanceArgs($params);
+            } else {
```

Comment:
Well if you remove it the code is less clearer imo and it's harder to understand this 2 lines...